Tools/jit/trampoline.c

```diff
@@ -10,7 +10,7 @@ _Py_CODEUNIT *
 _JIT_ENTRY(
     _PyExecutorObject *exec, _PyInterpreterFrame *frame, _PyStackRef *stack_pointer, PyThreadState *tstate
 ) {
-    typedef DECLARE_TARGET((*jit_func));
+    typedef _Py_CODEUNIT *__attribute__((preserve_none)) (*jit_func)(_PyInterpreterFrame *, _PyStackRef *, PyThreadState *);
     jit_func jitted = (jit_func)exec->jit_code;
```

Comment:
```suggestion
    jit_func_preserve_none jitted = (jit_func_preserve_none)exec->jit_code;
```
I suggest to use
https://github.com/python/cpython/blob/f937468e7c88c768a28ff4e653da051ffa30d86c/Tools/jit/jit.h#L1-L4
which builds on top of
https://github.com/python/cpython/blob/d86ad870cc112d4bd36165398368826746555730/Include/internal/pycore_jit.h#L18
and IMHO is better than duplicating?

Reply:
Looking at the other usage of `jit_func_preserve_none` in
https://github.com/python/cpython/blob/f937468e7c88c768a28ff4e653da051ffa30d86c/Tools/jit/template.c#L50-L56
might suggest that it is even possible to omit the cast in
```
jit_func_preserve_none jitted = (jit_func_preserve_none)exec->jit_code;
```
but that results in less optimized code for `emit_trampoline` in the generated stencils header.

Adding it to `TIER2_TO_TIER2` doesn't change the stencil code for `emit__EXIT_TRACE` and `emit__COLD_EXIT` (the only places where it is used atm), though [^1]. Most probably, because it is followed by `musttail`, but the cast wouldn't hurt there, either?

[^1]: Tested only for x86_64-pc-windows-msvc.

Reply:
Sorry for the back and forth, but having a closer look and [playing on Godbolt](https://godbolt.org/z/efEvjj8jP) suggests that the cast is better in case of `-Wpedantic`, since strictly speaking `void *jit_code`
https://github.com/python/cpython/blob/d86ad870cc112d4bd36165398368826746555730/Include/internal/pycore_optimizer.h#L52-L61
is a data pointer. So maybe change to a cast in `TIER2_TO_TIER2`, too?

Reply:
Hmm, I'm not really concerned about `-Wpedantic` since it's not enabled in the default build configuration. Building with `-Wpedantic` will produce many warnings throughout the codebase, not just for this code.

Reply:
Yeah, no need to be pedantic about existing code :)